### PR TITLE
Pan 16

### PR DESCRIPTION
**Pull request by tramanh305**
finished

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,149 +1,33 @@
 buildscript {
+    ext {
```
Comment (atusa17) [inline build.gradle]: Is there a reason you deleted most of this file? The dependencies you specified are already in the build.gradle file

File: out/production/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+spring.mvc.view.prefix:/WEB-INF/jsp/
```
Comment (atusa17) [inline out/production/resources/application.properties]: These lines should be added  to the `application.properties` file located in edu.msudenver.tsp.src.main.resources

File: persistence/build.gradle [hunk 3]
```diff
@@ -9,11 +9,7 @@ version '1.0'
 
 sourceCompatibility = 1.8
 
-sonarqube {
-    properties {
-        property "sonar.projectName", 'Pandamonium Persistence Tier'
-    }
-}
+
```
Comment (atusa17) [inline persistence/build.gradle]: This block should not have been deleted. The persistence module should not have been touched. Please revert this change.

File: services/build.gradle [hunk 4]
```diff
@@ -7,11 +7,7 @@ version '1.0'
 
 sourceCompatibility = 1.8
 
-sonarqube {
-    properties {
-        property "sonar.projectName", 'Theorem Prover Services'
-    }
-}
+
```
Comment (atusa17) [inline services/build.gradle]: This block should not have been deleted. The services module should not have been touched. Please revert this change.

File: src/main/java/edu/msudenver/tsp/website/Application.java [hunk 5]
```diff
@@ -1,23 +1,21 @@
...
 @SpringBootApplication
 public class Application {
 
     public static void main(final String[] args) {
         SpringApplication.run(Application.class, args);
     }
+}
 
-    @Bean
+    /*@Bean
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/Application.java]: As a convention, we don't want to leave commented out code in production code. Please delete this method if you're not going to be using it.

File: src/main/java/edu/msudenver/tsp/website/Application.java [hunk 6]
```diff
@@ -1,23 +1,21 @@
...
+
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.context.ApplicationContext;
-import org.springframework.context.annotation.Bean;
 
-import java.util.Arrays;
 
-@Slf4j
+//import lombok.extern.slf4j.Slf4j;
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/Application.java]: If you leave this line uncommented, your SaveActions will automatically remove the import from the file and you don't need to comment it out like this

File: utilities/build.gradle [hunk 7]
```diff
@@ -7,11 +7,7 @@ version '1.0'
 
 sourceCompatibility = 1.8
 
-sonarqube {
-    properties {
-        property "sonar.projectName", 'Parsing and Proofs Utilities'
-    }
-}
+
```
Comment (atusa17) [inline utilities/build.gradle]: This block should not have been deleted. The utilitiesmodule should not have been touched. Please revert this change.

File: build.gradle (hunk 1 above)
Comment (tramanh305) [reply build.gradle]: My bad, already fixed!

File: out/production/resources/application.properties (hunk 2 above)
Comment (tramanh305) [reply out/production/resources/application.properties]: I think its already there right

File: persistence/build.gradle (hunk 3 above)
Comment (tramanh305) [reply persistence/build.gradle]: Reverted! The reason was the persistence issues cant make my task to work, thats why

File: services/build.gradle (hunk 4 above)
Comment (tramanh305) [reply services/build.gradle]: Reverted!

File: src/main/java/edu/msudenver/tsp/website/Application.java (hunk 6 above)
Comment (tramanh305) [reply src/main/java/edu/msudenver/tsp/website/Application.java]: Fixed!

File: src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+
+@Controller
+public class TheoremEntryController {
+
+    @Autowired
+    ProofDriver proofDriver;
```
Comment (altus1) [inline src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java]: Make this variable private.

File: src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+
+@Controller
+public class TheoremEntryController {
+
+    @Autowired
```
Comment (altus1) [inline src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java]: Instead of using field autowiring, you can use the constructor autowiring. It's more secure than field autowiring. In fact, Lombok makes this really easy for you. Just get rid of the @Autowired annotation here, and instead at the top of the class where you have @Controller written, add @AllArgsContructor and Lombok will create the constructor for you and automatically autowire the dependencies.

File: src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+
+@Controller
+public class TheoremEntryController {
```
Comment (altus1) [inline src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java]: You need to add a @RequestMapping("/theorem") here so that Spring Boot knows what class to use when a request from the website comes in that ends the URL in localhost:8080/theorem

File: src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Autowired
+    ProofDriver proofDriver;
+
+    @RequestMapping("/welcome")
```
Comment (altus1) [inline src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java]: This particular method seems redundant since Ze is creating the homepage for PAN-18

File: src/main/java/edu/msudenver/tsp/website/forms/Theorem.java [hunk 12]
```diff
@@ -0,0 +1,19 @@
+package edu.msudenver.tsp.website.forms;
+
+public class Theorem {
+    private String theoremName ;
+
+
+    public String getTheoremName() {
```
Comment (altus1) [inline src/main/java/edu/msudenver/tsp/website/forms/Theorem.java]: Instead of manually writing the setters and getters, you can use lombok to do it for you! Simply go to the top of your class, and add the annotations @Getter and @Setter and Lombok will automatically generate the setters and getters for every field in your class for you.

File: src/main/java/edu/msudenver/tsp/website/service/ProofDriver.java [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class ProofDriver {
+    List<String> theromList = new ArrayList<String>();
+    public String processProof(String theoremName){
+        //Business Logic
```
Comment (altus1) [inline src/main/java/edu/msudenver/tsp/website/service/ProofDriver.java]: I'm not sure I understand what this is doing here.

File: src/main/webapp/WEB-INF/jsp/welcome.jsp [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<%@ page language="java" contentType="text/html; charset=ISO-8859-1"
```
Comment (altus1) [inline src/main/webapp/WEB-INF/jsp/welcome.jsp]: I'm going to guess this was from a SpringGuru's guide to making a webpage haha but this is redundant since Ze is already making a home page for PAN-18

File: src/main/webapp/WEB-INF/jsp/welcome.jsp [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
+<html>
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
+    <title>Guru Success Page</title>
+</head>
+<body>
+<a><b>Welcome User!!!!</b></a>
+</body>
+</html>
```
Comment (altus1) [inline src/main/webapp/WEB-INF/jsp/welcome.jsp]: I'm not seeing any test classes anywhere. Am I just missing them? And did you also test the webpage before you committed it?

File: src/main/java/edu/msudenver/tsp/website/Application.java [hunk 16]
```diff
@@ -9,25 +12,19 @@
 
 import java.util.Arrays;
 
-@Slf4j
 @SpringBootApplication
+@Slf4j
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/Application.java]: Since you deleted the method that used this annotation, you're free to delete the annotation from this class entirely.

File: src/main/java/edu/msudenver/tsp/website/forms/Theorem.java [hunk 17]
```diff
@@ -0,0 +1,26 @@
+package edu.msudenver.tsp.website.forms;
+
+public class Theorem {
+    private String theoremName1 ;
+    private String theoremName2 ;
+
+    public String getTheoremName1() {
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/forms/Theorem.java]: Instead of manually writing these, annotate the top of your class with @Getter and @Setter and Lombok will automatically generate your setters and getters for you.

File: src/main/java/edu/msudenver/tsp/website/forms/Theorem.java [hunk 18]
```diff
@@ -0,0 +1,26 @@
+package edu.msudenver.tsp.website.forms;
+
+public class Theorem {
+    private String theoremName1 ;
+    private String theoremName2 ;
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/forms/Theorem.java]: We're expecting theorems to only have one name for now, so this is great! But where is a user actually going to specify the theorem? Could you change one of these fields to contain the actual theorem?

File: src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+    public ModelAndView firstPage()
+    {
+        return new ModelAndView("welcome");
+    }
+
+    @RequestMapping("/theorem")
+    public ModelAndView theoremPage()
+    {
+        //System.out.println("Inside controller");
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java]: Can you delete this line? We don't want to leave commented out methods in the code.

File: src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+public class TheoremEntryController {
+
+
+    @RequestMapping("/welcome")
+    public ModelAndView firstPage()
+    {
+        return new ModelAndView("welcome");
+    }
+
+    @RequestMapping("/theorem")
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java]: Chance this to @GetMapping("/theorem").

File: src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+    public ModelAndView theoremPage()
+    {
+        //System.out.println("Inside controller");
+        return new ModelAndView("Theorem");
+    }
+
+    @RequestMapping(value = "/save", method = RequestMethod.POST)
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java]: Change this to @PostMapping("/save")

File: src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+    public String saveTheorem(@Validated Theorem theorem, Model model) {
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/TheoremEntryController.java]: You added validation on the Theorem form, but didn't specify what to validate. Could you add `@NotBlank(min = 1, message = "the theorem name must consist of at least one character")` to the `theoremName` field and similarly for the `theorem` field?

File: src/main/webapp/WEB-INF/jsp/Theorem.jsp [hunk 23]
```diff
@@ -0,0 +1,16 @@
...
+    pageEncoding="UTF-8"%>
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
+<html>
+<head>
+<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+<title>Theroem Page</title>
+</head>
+<body>
+<form action="save" method="post">
+Enter Theorem 1: <input type="text" name="theoremName1">
```
Comment (atusa17) [inline src/main/webapp/WEB-INF/jsp/Theorem.jsp]: Can you change this to `Enter Theorem Name`?

File: src/main/webapp/WEB-INF/jsp/Theorem.jsp [hunk 24]
```diff
@@ -0,0 +1,16 @@
...
+    <br>Enter Theorem 2: <input type="text" name="theoremName2"><br>
```
Comment (atusa17) [inline src/main/webapp/WEB-INF/jsp/Theorem.jsp]: Can you change this to `Enter Theorem`?

File: src/main/webapp/WEB-INF/jsp/success.jsp [hunk 25]
```diff
@@ -0,0 +1,7 @@
+<html>
+<body>
+<br><b>Name1:</b><%= request.getParameter("theoremName1")%>
```
Comment (atusa17) [inline src/main/webapp/WEB-INF/jsp/success.jsp]: Can you change this to `Theorem name:`?

File: src/main/webapp/WEB-INF/jsp/success.jsp [hunk 26]
```diff
@@ -0,0 +1,7 @@
...
+<br><b>Name2:</b><%= request.getParameter("theoremName2")%>
```
Comment (atusa17) [inline src/main/webapp/WEB-INF/jsp/success.jsp]: Can you change this to `Theorem`?

File: src/main/webapp/WEB-INF/jsp/welcome.jsp (hunk 15 above)
Comment (atusa17) [reply src/main/webapp/WEB-INF/jsp/welcome.jsp]: Did you commit your test classes as well?

File: src/main/java/edu/msudenver/tsp/website/controller/Application.java [hunk 27]
```diff
@@ -0,0 +1,23 @@
+package edu.msudenver.tsp.website.controller;
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/Application.java]: So this class needs to stay in the website package. Only the `TheoremEntryController` needed to be moved.

File: src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+@Slf4j
+@Controller
+@AllArgsConstructor
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java]: Annotate this class with @RequestMapping("/theorem")

File: src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java [hunk 29]
```diff
@@ -0,0 +1,43 @@
...
+public class TheoremEntryController {
+
+
+
+    @RequestMapping("/theorem")
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java]: Change this to @GetMapping("/")

File: src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java [hunk 30]
```diff
@@ -0,0 +1,43 @@
...
+    public ModelAndView theoremPage()
+    {
+
+
+        return new ModelAndView("Theorem");
+    }
+
+
+    @RequestMapping(value = "/save", method = RequestMethod.POST)
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java]: Change this to @PostMapping("/")

File: src/main/java/edu/msudenver/tsp/website/controller/forms/Theorem.java [hunk 31]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.Setter;
+
+@Getter
+@Setter
+
+public class Theorem {
+    private String theoremName ;
+
+
+    public String getTheoremName() {
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/forms/Theorem.java]: Since you've used the Lombok annotations, you don't need these methods anymore. Also, can you add a field for storing the Theorem itself too. So a field for the name of the theorem, and a field for the theorem itself.

File: src/main/webapp/WEB-INF/jsp/welcome.jsp [hunk 32]
```diff
@@ -0,0 +1,11 @@
+
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
+<html>
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
+    <title>Guru Success Page</title>
```
Comment (atusa17) [inline src/main/webapp/WEB-INF/jsp/welcome.jsp]: Can you rename this page to something like "Theorem Saved"?

File: src/main/webapp/WEB-INF/jsp/welcome.jsp [hunk 33]
```diff
@@ -0,0 +1,11 @@
...
+</head>
+<body>
+<a><b>Welcome User!!!!</b></a>
```
Comment (atusa17) [inline src/main/webapp/WEB-INF/jsp/welcome.jsp]: Can you change this message to something like "Successfully saved the theorem!"

File: src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java [hunk 34]
```diff
@@ -0,0 +1,43 @@
...
+
+@Slf4j
+@Controller
+@AllArgsConstructor
+public class TheoremEntryController {
+
+
+
+    @RequestMapping("/theorem")
+    public ModelAndView theoremPage()
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java]: Rename this method to be more descriptive of its purpose. So, something like "enterTheoremPage()"

File: src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java [hunk 35]
```diff
@@ -0,0 +1,43 @@
...
+    @RequestMapping(value = "/save", method = RequestMethod.POST)
+    public String saveTheorem(@Validated Theorem theorem, Model model) {
+
+        model.addAttribute("theromName", theorem.getTheoremName());
+
+        return "success";
+    }
+
+
+    public ModelAndView firstPage() {
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java]: Ze is already creating the welcome page, so this method is redundant. Can you remove this method?

File: src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java [hunk 36]
```diff
@@ -0,0 +1,53 @@
...
+import static org.junit.Assert.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TheoremEntryControllerTest {
+
+    @InjectMocks
+    private TheoremEntryController theoremEntryController;
+
+    @Autowired
+    protected MockMvc mockMvc;
```
Comment (atusa17) [inline src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java]: This variable is unused: can you remove it please?

File: src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java [hunk 37]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Test
+    public void contexLoads() throws Exception {
```
Comment (atusa17) [inline src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java]: This test isn't testing the `TheoremEntryController`, but rather JUnit and SpringJunit4Runer. So this test isn't testing what you want to test. Can you remove it?

File: src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java [hunk 38]
```diff
@@ -0,0 +1,53 @@
...
+        assertNotNull(theoremEntryController);
+    }
+
+    @Test
+    public void firstPage(){
```
Comment (atusa17) [inline src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java]: Can you remove this method since Ze is already making a home page?

File: src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java [hunk 39]
```diff
@@ -0,0 +1,53 @@
...
+
+        final ModelAndView modelAndView= theoremEntryController.firstPage();
+
+        assertNotNull(modelAndView);
+        assertEquals("welcome",modelAndView.getViewName());
+    }
+
+    @Test
+    public void theoremPage(){
+
```
Comment (atusa17) [inline src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java]: I know this is kind of knit-picky, but can you delete this extra space after you declare your method to when you start the test?

File: src/main/java/edu/msudenver/tsp/website/controller/forms/Theorem.java [hunk 40]
```diff
@@ -0,0 +1,25 @@
+package edu.msudenver.tsp.website.controller.forms;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+
+public class Theorem {
+    private String theoremName ;
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/forms/Theorem.java]: Can you put the annotation `@NotBlank` in front of this so that your `@Validation` usage in the `TheoremEntryController` actually validates that this field is not blank? And can you also delete the extra space after the field name and the semicolon?

File: src/main/java/edu/msudenver/tsp/website/controller/forms/Theorem.java [hunk 41]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/forms/Theorem.java]: Add another field for storing the actual theorem itself: `@NotBlank private String theorem;`

File: src/main/webapp/WEB-INF/jsp/welcome.jsp [hunk 42]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (atusa17) [inline src/main/webapp/WEB-INF/jsp/welcome.jsp]: Since Ze is already creating a homepage, you can delete this jsp file.

File: src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java [hunk 43]
```diff
@@ -0,0 +1,53 @@
...
+        final ModelAndView modelAndView= theoremEntryController.theoremPage();
+
+        assertNotNull(modelAndView);
+        assertEquals("Theorem",modelAndView.getViewName());
+    }
+
```
Comment (atusa17) [inline src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java]: Can you also add a test to test the `saveTheorem` method in the `TheoremEntryController`?

File: src/main/webapp/WEB-INF/jsp/Theorem.jsp [hunk 44]
```diff
@@ -0,0 +1,16 @@
...
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
+<html>
+<head>
+<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+<title>Theroem Page</title>
+</head>
+<body>
+<form action="save" method="post">
+Enter Theorem Name: <input type="text" name="theoremName">
+
```
Comment (atusa17) [inline src/main/webapp/WEB-INF/jsp/Theorem.jsp]: Can you also add a field for entering the theorem itself?

File: src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java [hunk 45]
```diff
@@ -0,0 +1,43 @@
...
+    {
+
+
+        return new ModelAndView("Theorem");
+    }
+
+
+    @RequestMapping(value = "/save", method = RequestMethod.POST)
+    public String saveTheorem(@Validated Theorem theorem, Model model) {
+
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java]: I know this is knit-picky, but can you delete this extra space after your method declaration?

File: src/main/java/edu/msudenver/tsp/website/controller/Application.java [hunk 46]
```diff
@@ -0,0 +1,23 @@
...
+
+
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/Application.java]: Delete these spaces after the `@SpringBootAnnotation` and the start of the `Application` class.

File: src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java [hunk 47]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+@Slf4j
+@Controller
+@AllArgsConstructor
+public class TheoremEntryController {
+
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java]: Delete these extra spaces after you start your class and when you start your methods.

File: src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java [hunk 48]
```diff
@@ -0,0 +1,43 @@
...
+
+    @RequestMapping("/theorem")
+    public ModelAndView theoremPage()
+    {
+
+
+        return new ModelAndView("Theorem");
+    }
+
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java]: It's best practice to only ever have at most 1 space between lines of code. Can you delete one of these lines?

File: src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java [hunk 49]
```diff
@@ -0,0 +1,43 @@
...
+@Controller
+@AllArgsConstructor
+public class TheoremEntryController {
+
+
+
+    @RequestMapping("/theorem")
+    public ModelAndView theoremPage()
+    {
+
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java]: It's best practice to only ever have at most 1 space between lines of code. Can you delete one of these lines?

File: src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java [hunk 50]
```diff
@@ -0,0 +1,43 @@
...
+    @RequestMapping(value = "/save", method = RequestMethod.POST)
+    public String saveTheorem(@Validated Theorem theorem, Model model) {
+
+        model.addAttribute("theromName", theorem.getTheoremName());
+
+        return "success";
+    }
+
+
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/TheoremEntryController.java]: It's best practice to only ever have at most 1 space between lines of code. Can you delete one of these lines?

File: src/main/java/edu/msudenver/tsp/website/controller/forms/Theorem.java [hunk 51]
```diff
@@ -0,0 +1,25 @@
...
+
+    public String getTheoremName() {
+        return theoremName;
+    }
+
+    public void setTheoremName1(String theoremName) {
+        this.theoremName = theoremName;
+    }
+
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/forms/Theorem.java]: Can you delete all of the whitespace between this bracket and the one that ends your class?

File: src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java (hunk 43 above)
Comment (atusa17) [inline src/test/java/edu/msudenver/tsp/website/controller/TheoremEntryControllerTest.java]: It's best practice to only ever have at most 1 space between lines of code. Can you delete some of these lines?

File: src/main/java/edu/msudenver/tsp/website/controller/Application.java [hunk 52]
```diff
@@ -0,0 +1,23 @@
...
+
+public class Application {
+
+    public static void main(final String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/Application.java]: Delete this space between the brackets.

File: src/main/java/edu/msudenver/tsp/website/controller/Application.java [hunk 53]
```diff
@@ -0,0 +1,23 @@
...
+}
+
```
Comment (atusa17) [inline src/main/java/edu/msudenver/tsp/website/controller/Application.java]: Delete all of these extra spaces after this bracket that closes your class.
